script.js

```diff
@@ -1,4 +1,542 @@
+    return `${player2}.<br>Your 1st Dice = ${player2Dice1}<br> Your 2nd Dice = ${player2Dice2}.<br><br>Enter '1' or '2' to choose the 1st or 2nd Dice  as your 1st Numeral respectively.`;
+  }
+};
+
+// Player choose Dice Order
+var chooseDiceOrder = function (playerOrderChoice) {
+  if (currentPlayer == player1 && playerOrderChoice == 1) {
+    player1Num = Number("" + player1Dice1 + player1Dice2);
+    console.log(player1Num);
+    return player1Num;
+  } else if (currentPlayer == player1 && playerOrderChoice == 2) {
+    player1Num = Number("" + player1Dice2 + player1Dice1);
+    console.log(player1Num);
+    return player1Num;
+  }
+  if (currentPlayer == player2 && playerOrderChoice == 1) {
+    player2Num = Number("" + player2Dice1 + player2Dice2);
+    console.log(player2Num);
+    return player2Num;
+  } else if (currentPlayer == player2 && playerOrderChoice == 2) {
+    player2Num = Number("" + player2Dice2 + player2Dice1);
+    console.log(player2Num);
+    return player2Num;
+  }
+};
+
+var generateCombinedNum = function () {
+  player1Dice1 = generateRandomDiceNumber();
+  player1Dice2 = generateRandomDiceNumber();
+  player2Dice1 = generateRandomDiceNumber();
+  player2Dice2 = generateRandomDiceNumber();
+  if (gameMode == 3) {
+    if (currentPlayer == player1) {
+      if (player1Dice1 >= player1Dice2) {
+        player1Num = Number("" + player1Dice1 + player1Dice2);
+        console.log(`auto ${player1Num}`);
+        return player1Num;
+      } else if (player1Dice1 < player1Dice2) {
+        player1Num = Number("" + player1Dice2 + player1Dice1);
+        console.log(`auto ${player1Num}`);
+        return player1Num;
+      }
+    }
+    if (currentPlayer == player2) {
+      if (player2Dice1 >= player2Dice2) {
+        player2Num = Number("" + player2Dice1 + player2Dice2);
+        console.log(`auto ${player2Num}`);
+        return player2Num;
+      } else if (player2Dice1 < player2Dice2) {
+        player2Num = Number("" + player2Dice2 + player2Dice1);
+        console.log(`auto ${player2Num}`);
+        return player2Num;
+      }
+    }
+  }
+  if (gameMode == 4) {
+    if (currentPlayer == player1) {
+      if (player1Dice1 >= player1Dice2) {
+        player1Num = Number("" + player1Dice2 + player1Dice1);
+        console.log(`auto ${player1Num}`);
+        return player1Num;
+      } else if (player1Dice1 < player1Dice2) {
+        player1Num = Number("" + player1Dice1 + player1Dice2);
+        console.log(`auto ${player1Num}`);
+        return player1Num;
+      }
+    }
+    if (currentPlayer == player2) {
+      if (player2Dice1 >= player2Dice2) {
+        player2Num = Number("" + player2Dice2 + player2Dice1);
+        console.log(`auto ${player2Num}`);
+        return player2Num;
+      } else if (player2Dice1 < player2Dice2) {
+        player2Num = Number("" + player2Dice1 + player2Dice2);
+        console.log(`auto ${player2Num}`);
+        return player2Num;
+      }
+    }
+  }
+};
```

Comment:
Since you are returning the number generated at the end of every if or else if statement (meaning that the code isn't reliant on any of the other code), and they are mostly similar, you can look into simply combining some of the statements. Also, it seems that the gameMode doesn't affect the order of the dice? I.E the code across the game modes seems the same.